docs/search-usage.rst

```diff
@@ -0,0 +1,215 @@
+   >>> field.add_value('string')
+
+Save the document into the index:
+
+.. doctest::
+
+   >>> document.create()  # API call
+   >>> document.exists()  # API call
+   True
+   >>> document.rank      # set by the server
+   1443648166
+
+List all documents in an index:
+
+.. doctest::
+
+   >>> documents = index.list_documents()  # API call
+   >>> [document.id for document in documents]
+   ['document-1']
+
+Delete a document from its index:
+
+.. doctest::
+
+   >>> document = index.document('to-be-deleted')
+   >>> document.exists()  # API call
+   True
+   >>> document.delete()  # API call
+   >>> document.exists()  # API clal
+   False
+
+.. note::
+
+   To update aocuments in place after manipulating its fields or rank, just
+   recreate it:  E.g.:
+
+   .. doctest::
+
+      >>> document = index.document('document-id')
+      >>> document.exists()  # API call
+      True
+      >>> document.rank = 12345
+      >>> field = document.field('field-name')
+      >>> field.add_value('christina aguilera')
+      >>> document.create()  # API call
+
+Fields
+~~~~~~
+
+Fields belong to documents and are the data that actually gets searched.
+
+Each field can have multiple values, which can be of the following types:
+
+- String (Python2 :class:`unicode`, Python3 :class:`str`)
+- Number (Python :class:`int` or :class:`float`)
+- Timestamp (Python :class:`datetime.datetime`)
+- Geovalue (Python tuple, (:class:`float`, :class:`float`)
+
+String values can be tokenized using one of three different types of
+tokenization, which can be passed when the value is added:
+
+- **Atom** (``atom``) means "don't tokenize this string", treat it as one
+  thing to compare against.
+
+- **Text** (``text``) means "treat this string as normal text" and split words
+  apart to be compared against.
+
+- **HTML** (``html``) means "treat this string as HTML", understanding the
+  tags, and treating the rest of the content like Text.
+
+.. doctest::
+
+   >>> from gcloud import search
+   >>> client = search.Client()
+   >>> index = client.index('index-id')
+   >>> document = index.document('document-id')
+   >>> field = document.field('field-name')
+   >>> field.add_value('britney spearks', tokenization='atom')
+   >>> field.add_value(''<h1>Britney Spears</h1>', tokenization='html')
+
+Searching
+~~~~~~~~~
+
+After populating an indexe with documents, search through them by
+issuing a search query:
+
+.. doctest::
+
+   >>> from gcloud import search
+   >>> client = search.Client()
+   >>> index = client.index('index-id')
+   >>> query = client.query('britney spears')
+   >>> matching_documents = index.search(query)  # API call
+   >>> for document in matching_documents:
+   ...     print document.id
+   ['document-id']
+
+By default, all queries are sorted by the ``rank`` value set when the
+document was created.  See:
+https://cloud.google.com/search/reference/rest/v1/projects/indexes/documents#resource_representation.google.cloudsearch.v1.Document.rank
```

Comment:
I already cleared this with the PM for cloud search as part of building it out on gcloud-node. We're all fine to do this -- no worries :)